README.md

```diff
@@ -34,16 +36,344 @@ version can be found on in the maven repository for
 </dependency>
 ```
 
-### Getting Started
+### Javadocs
 There are canonical examples defined in the class definition Javadoc of the client packages.
 
 Documentation can be found at the [Java Client
 Github Project Page](http://prometheus.github.io/client_java).
 
+
+## Instrumenting
+
+Four types of metrics are offered: Counter, Gauge, Summary and Histogram.
+See the documentation on [metric types](http://prometheus.io/docs/concepts/metric_types/)
+and [instrumentation best practices](http://prometheus.io/docs/practices/instrumentation/#counter-vs.-gauge,-summary-vs.-histogram)
+on how to use them.
+
+### Counter
+
+Counters go up, and reset when the process restarts.
+
+
+```java
+import io.prometheus.client.Counter;
+class YourClass {
+  static final Counter requests = Counter.build()
+     .name("requests_total").help("Total requests.").register();
+  
+  void processRequest() {
+    requests.inc();
+    // Your code here.
+  }
+}
+```
+
+### Gauge
+
+Gauges can go up and down.
+
+```java
+class YourClass {
+  static final Gauge inprogressRequests = Gauge.build()
+     .name("inprogress_requests").help("Inprogress requests.").register();
+  
+  void processRequest() {
+    inprogressRequest.inc();
+    // Your code here.
+    inprogressRequest.dec();
+  }
+}
+```
+
+There are utilities for common use cases:
+
+```java
+gauge.setToCurrentTime(); // Set to current unixtime.
+```
+
+As an advanced use case, a `Gauge` can also take its value from a callback. Keep in mind that the default `inc()`, `dec()` and `set()`
```

Comment:
> There is no example on Gauge in its JavaDocs and no one will be hunting in parent classes for the example.

Link to the example we have.

> I still can't wrap my head around why you think it shouldn't have up front documentation

Basically dropwizard has taught some developers that this is the way to handle gauges, whereas with Prometheus client libraries this is rarely the appropriate way to do things. Thus I want to make it less likely that users will tie themselves up in knots trying to use this dropwizard-style.